realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -409,15 +411,22 @@ private static RealmCache getCache(String realmPath, boolean createIfNotExist) {
      */
     static <E extends BaseRealm> E createRealmOrGetFromCache(RealmConfiguration configuration, Class<E> realmClass) {
         RealmCache cache = getCache(configuration.getPath(), true);
-        return cache.doCreateRealmOrGetFromCache(configuration, realmClass, OsSharedRealm.VersionID.LIVE);
+        boolean realmFileIsBeingCreated = !configuration.realmExists();
```

Comment:
This is not in a synchronized region with the `createInstance` call from within `cache.doCreateRealmOrGetFromCache` so would potentially be off.